kpops/component_handlers/streams_bootstrap/app_handler.py

```diff
@@ -21,35 +20,52 @@
 
 
 class AppHandler:
-    def __init__(self, helm_wrapper: HelmWrapper):
-        self._helm_wrapper = helm_wrapper
-        helm_wrapper.helm_repo_add()
+    def __init__(
+        self,
+        helm_config: HelmConfig,
+        helm_repo_config: HelmRepoConfig,
+        helm_diff: HelmDiff,
+    ):
+        self._helm_wrapper = Helm(helm_config)
+        self.repository_name = helm_repo_config.repository_name
+        self.chart_version = helm_repo_config.version
+        self._helm_wrapper.add_repo(
+            helm_repo_config.repository_name,
+            helm_repo_config.url,
+            RepoAuthFlags(
+                username=helm_repo_config.username, password=helm_repo_config.password
+            ),
+        )
+        self.helm_diff = helm_diff
 
     def install_app(
         self,
         release_name: str,
+        application_type: str,
         namespace: str,
         values: dict,
-        app_type: ApplicationType,
         dry_run: bool,
-        local_chart_path: Path | None = None,
     ):
         """
         Calls helm upgrade
         :param release_name: The release name of your chart
         :param namespace: The namespace where the chart is going to be released
         :param values: The value YAML for the chart
-        :param app_type: The type of the application
         :param dry_run: sets the --dry-run flag
         """
-        self._helm_wrapper.helm_upgrade_install(
+        stdout = self._helm_wrapper.upgrade_install(
             release_name=release_name,
+            chart=f"{self.repository_name}/{application_type}",
+            dry_run=dry_run,
             namespace=namespace,
             values=values,
-            app=app_type.value,
-            dry_run=dry_run,
-            local_chart_path=local_chart_path,
+            flags=HelmUpgradeInstallFlags(version=self.chart_version),
         )
+        if dry_run and self.helm_diff.config.enable:
```

Comment:
There should also be a way to make this diff reusable for custom Kubernetes apps